File: polyglot/Singleton/singleton.py
# Singleton design pattern

class SingleObject(object):
    mBuilt = False
    mInstance = None
    # can be called once only
    def __init__(self):
        if SingleObject.mBuilt == True:
            raise Exception("SingleObject: cannot create instance")
        SingleObject.mBuilt = True
    def getInstance():
        if SingleObject.mInstance is None:
            SingleObject.mInstance = SingleObject()
        return SingleObject.mInstance
    def showMessage(self):
        print("The single object is alive and well")
    

obj = SingleObject.getInstance()
obj.showMessage()


        
 

File: polyglot/Singleton/Singleton.java
// Singleton design pattern

class SingleObject {
  // private constructor
  private SingleObject(){}
  // static data member pointing to unique instance
  private static SingleObject mInstance = null;
  // get the only object available
  public static SingleObject getInstance(){
    if(mInstance == null){
      mInstance = new SingleObject();
    }
    return mInstance;
  }
  public void showMessage(){
    System.out.println("The single object is alive and well");
  }
 }

public class Singleton {
  public static void main(String[] args){
    // will not compile
    // SingleObject nope = new SingleObject();

    SingleObject object = SingleObject.getInstance();
    object.showMessage();
  }
}

